include/config/item.h

```diff
@@ -16,6 +16,10 @@
 // TM config
 #define I_REUSABLE_TMS          FALSE       // In Gen5-8, TMs are reusable. Setting this to TRUE will make all vanilla TMs reusable, though they can also be cherry-picked by setting their importance to 1.
 
+// Exp. Share config
+#define I_EXP_SHARE_FLAG        0           // If this flag is set, every Pokémon in the party will gain experience, regardless if they participated in the battle or not.
+#define I_EXP_SHARE_ITEM        GEN_5       // In Gen6+, the Exp. Share was changed from a held item to a Key item that toggles the effect described above.
```

Comment:
Shouldn't this be GEN_LATEST by default? I know a few items which classifications were changed by default, like the fossils (key item to regular item), so that is within scope here.

Reply:
Static assertion to avoid setting the Gen 6 item without the flag